web/js/app.js

```diff
@@ -376,25 +389,285 @@ LR.utils.preSelectMultiSelects = function(annotationData, keys, form){
+                    }  
+                })
+            }
+            else{
+                //This is disconcerning.  The deer-view either didn't load or the DOM didn't draw it fast enough...
+                console.warn("Could not pre-select multi selects.  The deer-view either didn't load or the DOM didn't draw it fast enough!"
+                        +"  A multi select may not be preselected.")
             }
-            data_arr.forEach(val => {
-                let option = sel.querySelector("option[value='"+val+"']")
-                if(option){
-                    option.selected = true
-                }
-                else{
-                    //The <option> is not available in the <select> HTML.
-                }  
-            })
         }
         else{
             //There is no annotation data for this key.
             console.warn("LR App tried to find '"+key+"' in this form data and could not.  A multi select may not be preselected.")
         }
-        
+    })
+}
+
+/**
+ * 
+ * Type and AdditionalType information has a custom UI around it, so DEER cannot prefill those pieces.
+ * Here we have the object with the type  
+ * @param {Object} annotationData The expanded containing all annotation data for a form.
+ * @param {Array(String)} keys The specific annotations we are looking for in annotationData
+ * @param {HTMLElement} form The completely loaded HTML <form> containing the <selects>s
+ * @return None
+ */
+LR.utils.preSelectType = function(object, form){
+    let type = object.hasOwnProperty("additionalType") ? object.additionalType : ""
+    type = type[0] // It is saving twice right now, so this handles the bug for DEMO purposes.  TODO FIXME
+    if(type){
+        var data_key_elements = form.querySelectorAll("[data-rdf]")
+        Array.from(data_key_elements).forEach(el => {
+            el = el.closest("[data-rdf]")
+            if (el.getAttribute("data-rdf") === type) {
+                el.classList.add("bg-light")
+            } 
+            else {
+                el.classList.remove("bg-light")
+            }
+        })
+        form.querySelector(".otherPrimaryTypes").value = type
+    }
+    else{
+        //There is no annotation data for this key.
+        console.warn("This object did not have a primary type!")
+        console.log(object)
+    }
+}
+
+/**
+ * 
+ * Type and AdditionalType information has a custom UI around it, so DEER cannot prefill those pieces.
+ * Here we have the object with the type  
+ * @param {Object} annotationData The expanded containing all annotation data for a form.
+ * @param {Array(String)} keys The specific annotations we are looking for in annotationData
+ * @param {HTMLElement} form The completely loaded HTML <form> containing the <selects>s
+ * @return None
+ */
+LR.utils.populateCoordinates = function(object, form){
+    let geo = object.hasOwnProperty("geometry") ? object.geometry : {}
```

Comment:
Is this the right way to do this? Have you consulted the IIIF Cookbook?

Reply:
Nope, but that's how they're built right now.  We will deal with it eventually #112